File: 5.implementation/redsun/src/main/webapp/static/js/services/VersionsService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('versionService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(version) {
			var urlServer = $rootScope.makePostURL("/version/insert");
			return $http.post(urlServer, version);
		}
		
		// Update.
		, update: function(versionId, version) {
			var urlServer = $rootScope.makePostURL("/version/update/" + versionId);
			return $http.put(urlServer, version);
		}
		
		// Delete.
		, delete: function(versionId) {
			var urlServer = $rootScope.makePostURL("/version/delete/" + versionId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(versionId) {
			var url = $rootScope.makeGetURL("/version/getbyid/" + versionId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, version) {
			var url = $rootScope.makeGetURL("/version/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: version });
		}
		
	}// return.
});

File: 5.implementation/VT/redsun-service/src/main/java/com/redsun/service/service/VersionsService.java
package com.redsun.service.service;

import com.redsun.service.entities.Versions;
import com.redsun.service.entities.Result;

/**
 * Versions Service interface
 */
public interface VersionsService {
	
    // Insert.
	Result insert(final Versions versions) ;

    // Update.
	Result update(final Versions versions) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listVersionsForPageAndFilter(final int itemsPerPage, final int pageNo, final Versions versions);

}
